tests/test_shipyard.py

```diff
@@ -0,0 +1,434 @@
+"""
+@file test_shipyard.py
+
+Unit tests for the Shipyard.py
+"""
+import pytest
+from PyQt5.QtWidgets import QApplication
+from PyQt5.QtTest import QTest
+from PyQt5.QtCore import Qt
+from imperium.shipyard import Window
+import sys
+
+
+
+
+
+def test_init():
+    # Test init of the window and its starting values
+    app = QApplication(sys.argv)
+
+    window = Window()
+    assert window is not None
+    assert window.spacecraft.tonnage == 0
+    assert window.logger.text() == ""
+    assert window.tonnage_line_edit.text() == "0"
+    assert window.cargo_line_edit.text() == "0"
+    assert window.fuel_line_edit.text() == "0"
+    assert window.jump_line_edit.text() == "0"
+    assert window.jump_label.text() == "-"
+    assert window.thrust_line_edit.text() == "0"
+    assert window.thrust_label.text() == "-"
+    assert window.hull_hp_line_edit.text() == "0"
+    assert window.structure_hp_line_edit.text() == "0"
+    assert window.armour_line_edit.text() == "0"
+    assert window.cost_line_edit.text() == "0.0"
+
+
+def test_edit_tonnage():
+    # Test setting tonnage and checking GUI updates
+    app = QApplication(sys.argv)
+
+    window = Window()
```

Comment:
Consolidate windows instantiation into a "before_all" type; else this stacks windows in memory, and is bad code reuse.